File: app/src/main/java/app/android/ttp/mikazuki/yoshinani/binding/BindableInt.java
package app.android.ttp.mikazuki.yoshinani.binding;

import android.databinding.BaseObservable;
import android.support.annotation.Nullable;

import org.parceler.Parcel;

import java.util.Objects;

/**
 * Created by haijimakazuki on 16/01/26.
 */
@Parcel
public class BindableInt extends BaseObservable {

    int mValue;

    public int get() {
        return mValue;
    }

    public void set(@Nullable final int value) {
        if (!Objects.equals(mValue, value)) {
            mValue = value;
            notifyChange();
        }
    }
}

File: app/src/main/java/app/android/ttp/mikazuki/yoshinani/binding/BindableInt.kt
package app.android.ttp.mikazuki.yoshinani.binding

import android.databinding.BaseObservable

import org.parceler.Parcel

import java.util.Objects

/**
 * Created by haijimakazuki on 16/01/26.
 */
@Parcel
class BindableInt : BaseObservable() {

    internal var mValue: Int = 0

    fun get(): Int {
        return mValue
    }

    fun set(value: Int) {
        if (mValue != value) {
            mValue = value
            notifyChange()
        }
    }
}